numcodecs/compat_ext.pxd

```diff
@@ -1,12 +1,11 @@
 # cython: language_level=3
 
 
-cdef class Buffer:
-    cdef:
-        char *ptr
-        Py_buffer buffer
-        size_t nbytes
-        size_t itemsize
-        bint acquired
-
-    cpdef release(self)
+cdef extern from *:
+    """
+    #define PyBytes_RESIZE(b, n) _PyBytes_Resize(&b, n)
+    """
```

Comment:
This should have an `#ifndef` guard to protect against redefinition error

So far it seems Cython doesn't encounter this error as it defines this once per module. Still it is a good idea to add the guard

Handling in PR: https://github.com/zarr-developers/numcodecs/pull/732